pallets/xyk/src/mock.rs

```diff
@@ -145,6 +145,7 @@ impl Config for Test {
 	type NativeAssetId = NativeAssetId;
 	type WeightInfo = ();
 	type GetExchangeFee = ExchangeFeeRate;
+	type AMMHandler = ();
```

Comment:
I'd consider splitting into separate handlers : OnCreatePool, OnTrade. 

Reason being that in future we'd like to make this pallet more generic and there can be a case where only one of them is needed.